packages/firestore/src/remote/remote_store.ts

```diff
@@ -661,11 +678,17 @@ export class RemoteStore implements TargetMetadataProvider {
     );
     const batch = this.writePipeline.shift()!;
     const success = MutationBatchResult.from(batch, commitVersion, results);
-    return this.syncEngine.applySuccessfulWrite(success).then(() => {
-      // It's possible that with the completion of this mutation another
-      // slot has freed up.
-      return this.fillWritePipeline();
-    });
+    try {
```

Comment:
It almost seems like this try/catch blurb could be its own method that take the operation to perform, tries it, and then passes it to `disableNetworkUntilRecovery`.

Reply:
Added a small wrapper. It is only used twice, but the wrapper itself is one one line, which is kind of nice.